config/eventhub/config.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2022 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package eventhub
+
+import (
+	"github.com/crossplane/terrajet/pkg/config"
+
+	"github.com/crossplane-contrib/provider-jet-azure/config/common"
+)
+
+// Configure configures resource group
+func Configure(p *config.Provider) {
+	p.AddResourceConfigurator("azurerm_eventhub_namespace", func(r *config.Resource) {
+		r.Version = common.VersionV1Alpha2
+		r.Kind = "EventHubNamespace"
+		r.UseAsync = true
+		r.ExternalName = config.NameAsIdentifier
+		r.ExternalName.GetExternalNameFn = common.GetNameFromFullyQualifiedID
+		// /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Eventhub/namespaces/namespace1
+		r.ExternalName.GetIDFn = common.GetFullyQualifiedIDFn("Microsoft.Eventhub",
+			"namespaces", "name",
+		)
+		r.LateInitializer = config.LateInitializer{
+			IgnoredFields: []string{"network_rulesets"},
+		}
+	})
+
+	p.AddResourceConfigurator("azurerm_eventhub", func(r *config.Resource) {
+		r.Version = common.VersionV1Alpha2
+		r.Kind = "EventHub"
+		r.References = config.References{
+			"namespace_name": config.Reference{
+				Type: "EventHubNamespace",
+			},
+		}
+		r.UseAsync = true
+		r.ExternalName = config.NameAsIdentifier
+		r.ExternalName.GetExternalNameFn = common.GetNameFromFullyQualifiedID
+		// /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.EventHub/namespaces/namespace1/eventhubs/eventhub1
+		r.ExternalName.GetIDFn = common.GetFullyQualifiedIDFn("Microsoft.Eventhub",
+			"namespaces", "name",
```

Comment:
Shouldn't this be:
```suggestion
			"namespaces", "namespace_name",
```

I believe we might have missed this because both the referenced `EventHubNamespace` and the `EventHub` itself are named as `example`.

Reply:
@ulucinar Actually I noticed that and double-checked today morning with *different* non-matching resource names. I expected it to fail but it worked for some reason. I will fix it but it would be great to understand why it didn't misbehave

Reply:
just for illustration even when just `name` specified everwhere a proper FQDN is getting constructed
```
id: /subscriptions/038f2b7c-3265-43b8-8624-c9ad5da610a8/resourceGroups/event-example/providers/Microsoft.EventHub/namespaces/yury-enamespace-example/eventhubs/yury-eventhub-example/consumergroups/yury-consumergroup-example
```
But I am not sure why :D